services/jvm/src/main/scala/io/gearpump/services/RestServices.scala

```diff
@@ -108,7 +105,11 @@ object RestServices {
 
     implicit val materializer = ActorMaterializer()
 
-    Http().bindAndHandle(Route.handlerFlow(services.routes), host, port)
+    Http().bindAndHandle(Route.handlerFlow(services.routes), host, port) onFailure {
+      case t: Throwable =>
+        System.err.println(t)
+        System.exit(-1)
```

Comment:
System.exit is too disruptive. It is dangerous to use it in YARN. Can you throw an exception instead?